test/ERC1155Tradable.js

```diff
@@ -0,0 +1,390 @@
+         assert.ok(supply.eq(toBN(100)));
+       });
+
+    // We check some of this in the other create() tests but this makes it
+    // explicit and is more thorough.
+    it('should set tokenSupply on creation',
+       async () => {
+         tokenId += 1;
+         truffleAssert.eventEmitted(
+           await instance.create(owner, 33, "", "0x0", { from: owner }),
+           'TransferSingle',
+           { _id: toBN(tokenId) }
+         );
+         const balance = await instance.balanceOf(owner, tokenId);
+         assert.ok(balance.eq(toBN(33)));
+         const supply = await instance.tokenSupply(tokenId);
+         assert.ok(supply.eq(toBN(33)));
+         assert.ok(supply.eq(balance));
+       });
+
+    it('should increment the token type id',
+       async () => {
+         // We can't check this with an accessor, so we make an explicit check
+         // that it increases in consecutive creates() using the value emitted
+         // in their events.
+         tokenId += 1;
+         await truffleAssert.eventEmitted(
+           await instance.create(owner, 0, "", "0x0", { from: owner }),
+           'TransferSingle',
+           { _id: toBN(tokenId) }
+         );
+         tokenId += 1;
+         await truffleAssert.eventEmitted(
+           await instance.create(owner, 0, "", "0x0", { from: owner }),
+           'TransferSingle',
+           { _id: toBN(tokenId) }
+         );
+       });
+
+    it('should not allow a non-owner to create tokens',
+       async () => {
+         truffleAssert.fails(
+           instance.create(userA, 0, "", "0x0", { from: userA }),
+           truffleAssert.ErrorType.revert,
+           'caller is not the owner'
+         );
+       });
+
+    it('should allow the contract owner to create tokens and emit a URI',
+       async () => {
+         tokenId += 1;
+         truffleAssert.eventEmitted(
+           await instance.create(owner, 0, URI_ROOT, "0x0", { from: owner }),
+           'URI',
+           {
+             _uri: URI_ROOT,
+             _id: toBN(tokenId)
+           }
+         );
+       });
+
+    it('should not emit a URI if none is passed',
+       async () => {
+         tokenId += 1;
+         truffleAssert.eventNotEmitted(
+           await instance.create(owner, 0, "", "0x0", { from: owner }),
+           'URI'
+         );
+       });
+  });
+
+  describe('#totalSupply()', () => {
+    it('should return correct value for token supply',
+       async () => {
+         tokenId += 1;
+         await instance.create(owner, 100, "", "0x0", { from: owner });
+         const balance = await instance.balanceOf(owner, tokenId);
+         assert.ok(balance.eq(toBN(100)));
+         // Use the created getter for the map
+         const supplyGetterValue = await instance.tokenSupply(tokenId);
+         assert.ok(supplyGetterValue.eq(toBN(100)));
+         // Use the hand-crafted accessor
+         const supplyAccessorValue = await instance.totalSupply(tokenId);
```

Comment:
this is the same code as the created getter scenario above, right?

Reply:
Sorry, how do you mean? :-)

Reply:
On line 177, it says `await instance.tokenSupply(tokenId);` on line 180, it says `await instance.totalSupply(tokenId);`. I don't see how they're different.

Reply:
Oh I see. :-) Yes, they *should* be the same as they expose the same information, so the test checks that.

The tokenSupply map should probably not be public in ERC1155Tradeable.sol given that the hand-crafted totalSupply() does return the same information.

I can add the assumptions behind the test to the comments, or change ERC1155Tradeable and remove the tokenSupply test. Let me know which you would prefer.

Reply:
Ohhhhh I just noticed the difference between the words `tokenSupply` and `totalSupply`, was so hard to see for some reason. nvm